osi3trace/osi_trace.py

```diff
@@ -49,13 +50,16 @@ def from_file(self, path, type_name="SensorView", cache_messages=False):
         self.message_offsets = [0]
         self.message_cache = {} if cache_messages else None
 
+        self.timestep_count = len(self.retrieve_offsets())
+        self.type = self.map_message_type(type_name)
+
     def retrieve_offsets(self, limit=None):
         """Retrieve the offsets of the messages from the file."""
         if not self.read_complete:
             self.current_index = len(self.message_offsets) - 1
             self.file.seek(self.message_offsets[-1], 0)
-        while (
-            not self.read_complete and not limit or len(self.message_offsets) <= limit
+        while not self.read_complete and (
+            not limit or len(self.message_offsets) <= limit
```

Comment:
The bugfix is important,

It just makes me regret that we added black reformatting checks that this is convoluted by the arbitrary re-indentations needed. We probably should increase the line length to avoid stupidity like this.